src/fia/fia/factor_instance_accumulator.rs

```diff
@@ -0,0 +1,119 @@
+use super::*;
+use crate::prelude::*;
+
+/// === FIA ===
+
+pub struct FactorInstanceAccumulator<ID, Path, Product>
```

Comment:
The generic types of `FIA` is:
1. `ID`: for signing this is `IntentHash`, for key derivation it can be `SecurityShieldID` or any `Uuid` for single key (might not be used)
2. `Path`: The type from which we can read out `DerivationPath` used to derive perform the operation. It is `HDPublicKey` for signing (as to validate derived key using _known_ public key) and simply `DerivationPath` (we `impl HasDerivationPath for DerivationPath`) for key derivation
3. `Product`: The outputs / products from usage of each factor source, for signing it is `HDSignature` and for key derivation it is `HDPublicKey` (we `impl HasHDPublicKey for HDPublicKey`).